### ISSUE TITLE
Qubes 4.1.1 - resize apps windows in sys-usb and other VMs freezes that VM

### ISSUE THREAD

**Issue by esatravi** [QubesOS/qubes-issues#7652]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release

4.1.1.

### Brief summary

Very strange yet repeatable behaviour in sys-usb. When you open the file manager (nautilus) in sys-usb and try to resize the window to make it bigger or full-screen, the sys-usb freezes and stops responding. It seems that after a long time (minutes) the window will get responsive again. 
I observed the same issue with other apps,like terminal - but here the freeze time is shorter. 

What is the dependency between sys-usb and windows resizing? 
Just for the information the sys-usb was installed with its default setting as disposable vm. 

Additional comments:
I've just tested other VMs, e.g. sys-net vm for this issue and it seems to be there as well. 

The window-resize freeze issue has certainly been introduced in Qubes 4.1.1
I tested new Qubes 4.1.1 installation on the other computer, the one I've been using without any problems for years with qubes 4.0. Unfortunately it now shows the same freeze symptoms when resizing windows. 

### Steps to reproduce

1. start sys-usb (or wait till it starts after rebooting)
2. Go to the application menu, select "System: sys-usb" -> "sys-usb: Files" to open nautilus file browser. 
3. When the files window appears drag it by the side or the corner and resize. You can also try to maximize it to full screen. In both cases the effect is the same - sys-usb vm freezes and become unresponsive. 
4. After few minutes sys-usb unfreezes but it seems it does not respond to typing/clicking as quickly. 

(5) I've tested other VMs, e.g. sys-net vm for this issue and it seems to be there as well. 

Summing up - there is something very strange going on on the 4.1.1 fresh installation - resizing file explorer / terminal windows freezes VM. 

### Expected behavior

Resizing any window within VM should be instantaneous, without freezing the entire app vm. 

### Actual behavior

see "steps-to-reproduce" section.

**Comment by esatravi** [comment]
The window-resize freeze issue has been introduced in Qubes 4.1.1. 

I tested new Qubes 4.1.1 installation on the other computer, the one I've been using without any problems for years with qubes 4.0. Unfortunately it shows the same freeze symptoms when resizing windows.

**Comment by DemiMarie** [comment]
Is it just the GUI that freezes?  Can you run a command in the VM using `qvm-run --no-gui`?

**Comment by esatravi** [comment]
I'm not quite sure what and how do you want to me to test this command. Please explain and provide and example and I will run them in the 

All I'm observing is that resizing windows of programs like nautilus or terminal causes the freeze of that vm. 

The more, the problem is new, introduced in 4.1.1 - I installed the latest qubes on the older computer, which I used over the past 5 years, and the problem exists there as well. 

I'm happy to help with testing but please provide the exact scenarios.

**Comment by DemiMarie** [comment]
This is likely a dom0 kernel problem and will be fixed by upgrading to 5.15.56 or better in dom0.  Assigning @marmarek.

**Comment by esatravi** [comment]
I have already tried to verify whether this is a kernel issue and upgraded it to the kernel-latest. When I execute 'uname -r' in dom0 I get the following readings: 5.18.9-1.fc32.qubes.x86_64 

That means it might not be kernel problem but something else. 

As I mentioned previously I installed qubes 4.1.1 on the older computer, where previous versions of qubesos were working without any issue for years. But the introduction of the 4.1.1. exhibits the same problems on both devices. 

4.1.1 changes the way GUI is handled and I suspect that introduced this new freezing issue. 

As I devised it doesn't seem to be related to the kernel version. Could you explore other possibilities?

**Comment by DemiMarie** [comment]
@esatravi: That has https://github.com/torvalds/linux/commit/dbe97cff7dd9f0f75c524afdd55ad46be3d15295 but not https://github.com/torvalds/linux/commit/166d3863231667c4f64dee72b77d1102cdfad11f, so you might experience a slowdown due to WARN() storms.

**Comment by esatravi** [comment]
@DemiMarie 
How do I revert the kernel-latest to the regular qubes LTS version (currently 5.15) please?

**Comment by DemiMarie** [comment]
> @DemiMarie Thank you for the tip. I've restarted Qubes using LTS 5.15 kernel. I was then able to remove the kernel-latest with dnf remove command.

The most recent `kernel-latest` packages (in `current-testing`) should have that fix too.

**Comment by esatravi** [comment]
@DemiMarie 
Are you aware of others having similar issue with the latest Qubes 4.1.1 when application windows resize causes vm freezing? 

Regarding the fix you mentioned - I had previously installed and tested kernel-latest (5.18.9-1.fc32.qubes.x86_64) but it didn't resolve the freezing issue? 
As of now kernel-latest' stil refers to v 5.18.9. 

What other things could potentially cause the freezing behavior of VMs?

**Comment by DemiMarie** [comment]
@esatravi you will need to enable testing repositories to get 5.18.14.

**Comment by esatravi** [comment]
@DemiMarie Thank you for pointing that out. 

How long will it take for that testing kernel version (5.18.14) to become and official version supported by QubesOS? When can I expect it go live? 

I'm not really keen on using the dev version of the kernel with all the warning enabled for debug purposes.

**Comment by DemiMarie** [comment]
@esatravi The package in `current-testing` is the same package that will eventually be moved to `current`.  It is fully security supported and is not a debug kernel.  I at least have the testing repositories enabled in all qubes on my daily driver machine, and I would not be surprised if @marmarek does too.

**Comment by esatravi** [comment]
@DemiMarie 
I've installed the current-testing version of the kernel v 5.18.14 but unfortunately the resize applications windows issue still persists. (confirmed with dom0: uname -r after reboot)
Perhaps the cause is not related to the kernel but rather something different, a bug in qubesos? It certainly needs to be investigated as it affects resizing in all VMs.

**Comment by DemiMarie** [comment]
@esatravi thanks!  yeah, might be a Qubes bug then, or possibly a driver bug.
